src/mono/wasm/runtime/loader/libraryInitializers.ts

```
@@ -0,0 +1,86 @@
+            const initializer = await import(/* webpackIgnore: true */ adjustedPath);
+
+            config.libraryInitializers!.push(initializer);
+        } catch (error) {
+            mono_log_warn(`Failed to import library initializer '${path}': ${error}`);
+        }
+    }
+}
+
+export async function invokeOnRuntimeConfigLoaded(config: MonoConfig) {
+    await fetchInitializers(config, "onRuntimeConfigLoaded");
+
+    if (!config.libraryInitializers) {
+        return;
+    }
+
+    const promises = [];
+    for (let i = 0; i < config.libraryInitializers.length; i++) {
+        const initializer = config.libraryInitializers[i] as { onRuntimeConfigLoaded: (config: MonoConfig) => Promise<void> };
+        if (initializer.onRuntimeConfigLoaded) {
+            promises.push(logAndSwallowError("onRuntimeConfigLoaded", () => initializer.onRuntimeConfigLoaded(config)));
+        }
+    }
+
+    await Promise.all(promises);
+    if (promises.length > 0)
+        normalizeConfig();
+}
+
+export async function invokeOnRuntimeReady(api: RuntimeAPI) {
+    const config = api.getConfig();
+    await fetchInitializers(config, "onRuntimeReady");
+
+    if (!config.libraryInitializers) {
+        return;
+    }
+
+    const promises = [];
+    for (let i = 0; i < config.libraryInitializers.length; i++) {
+        const initializer = config.libraryInitializers[i] as { onRuntimeReady: (api: RuntimeAPI) => Promise<void> };
+        if (initializer.onRuntimeReady) {
+            promises.push(logAndSwallowError("onRuntimeReady", () => initializer.onRuntimeReady(api)));
+        }
+    }
+
+    await Promise.all(promises);
+}
+
+async function logAndSwallowError(methodName: string, callback: () => Promise<void> | undefined): Promise<void> {
+    try {
+        await callback();
+    } catch (error) {
+        mono_log_warn(`Failed to invoke '${methodName}' on library initializer: ${error}`);
```

I think we should propagate the error